share/nisar/examples/noise_spectra_analysis.py

```diff
@@ -0,0 +1,448 @@
+        adding this threshold margin to it.
+    num_fft: int
+        Number of FFT as well number of samples per segment for the Welch Function.
+        Default: 2048
+    """
+
+    # Output file name is consisted of input file name + SPECTRA.hdf5
+    input_filename = os.path.basename(input_file)
+    input_base, input_ext = os.path.splitext(input_filename)
+    output_file = os.path.join(output_dir, f'{input_base}_SPECTRA{input_ext}')
+    
+    # Print output file information
+    print(f'{output_file = }')
+    print()
+
+    with h5py.File(output_file, 'w') as f_out:
+        # Write metadata as HDF5 attributes
+        f_out.attrs['num_pulses_block'] = num_pulses_blk
+        f_out.attrs['num_fft'] = num_fft
+        f_out.attrs['psd_x_axis'] = 'range (axis=1)'
+        f_out.attrs['psd_y_axis'] = 'power spectra density (axis=0, dB/Hz)'
+
+        for freq, pol_list in raw.polarizations.items(): # A, B
+            for pol in pol_list: # HH, HV, VV, VH
+                print(f'Start Processing: frequency{freq} {pol}')
+
+                # Read raw data
+                raw_data = raw.getRawDataset(freq, pol)
+                
+                pol_tx = pol[0]
+                fc, fs, _, _ = raw.getChirpParameters(freq, pol_tx)
+
+                # Estimate noise power of each dataset corresponding to a polarization
+                # EVD is used due to unknown RFI situation as a priori
+                noise_pwr_pulse_est_db = estimate_noise_pwr_pulse(raw_data, num_pulses_proc) 
+
+                # Convert Time-Domain Pulse Power to Frequency Domain Threshold in dB/Hz
+                threshold_db_hz = noise_pwr_pulse_est_db - 10*np.log10(fs) + thresh_margin
+
+                num_pulses, _ = raw_data.shape
+                slices = list(slice_gen(num_pulses, num_pulses_blk, combine_rem=False))
+                num_az_blocks = len(slices)
+
+                # Compute Frequencies from dummy data: one pulse
+                freqs, _ = welch(
+                    raw_data[0], 
+                    fs=fs, 
+                    nperseg=num_fft, 
+                    nfft=num_fft, 
+                    detrend=False, 
+                    return_onesided=False
+                )
+
+                # Shift to RF Center Frequency
+                freqs = fftshift(freqs) + fc
+                num_freq_bins = len(freqs)
+
+                # Create output group for output HDF5
+                base_path = f"/science/LSAR/QA/data/frequency{freq}/{pol}"
+                out_grp = f_out.require_group(base_path)
+
+                # Center Frequency
+                if "centerFrequency" in out_grp:
+                    del out_grp["centerFrequency"]
+                dset_fc = out_grp.require_dataset('centerFrequency', (), np.float64)
+                dset_fc[()] = fc
+                dset_fc.attrs["description"] = (
+                    f"Raw data center frequency"
+                )
```

Comment:
```suggestion
                dset_fc.attrs["description"] = np.bytes_(
                    f"Raw data center frequency"
                )
```
It looks like you updated the units but not the description attributes. Not a huge deal.

Reply:
@bhawkins , I addressed your comments with regards to block size and description.  I tested it without any issues.

thanks,
Bo